From 8e8e5451fc4c34fd3bb30a193be8499d5148898f Mon Sep 17 00:00:00 2001
Message-Id: <8e8e5451fc4c34fd3bb30a193be8499d5148898f@dist-git>
From: Martin Kletzander <mkletzan@redhat.com>
Date: Fri, 4 Nov 2016 10:29:40 +0100
Subject: [PATCH] qemu: Reorder shmem params nicely

Always format id first so that we don't need to do that twice in
different code paths.

Signed-off-by: Martin Kletzander <mkletzan@redhat.com>
(cherry picked from commit 6b5622e4b546eb0d0169b7f07ad3dae7a6560a35)

https://bugzilla.redhat.com/show_bug.cgi?id=1392031

Signed-off-by: Martin Kletzander <mkletzan@redhat.com>
---
 src/qemu/qemu_command.c                        |  6 ++++--
 tests/qemuxml2argvdata/qemuxml2argv-shmem.args | 16 ++++++++--------
 2 files changed, 12 insertions(+), 10 deletions(-)

diff --git a/src/qemu/qemu_command.c b/src/qemu/qemu_command.c
index 6eafa5e..3ebfe84 100644
--- a/src/qemu/qemu_command.c
+++ b/src/qemu/qemu_command.c
@@ -8471,13 +8471,15 @@ qemuBuildShmemDevLegacyStr(virDomainDefPtr def,
     }
 
     virBufferAddLit(&buf, "ivshmem");
+    virBufferAsprintf(&buf, ",id=%s", shmem->info.alias);
+
     if (shmem->size)
         virBufferAsprintf(&buf, ",size=%llum", shmem->size >> 20);
 
     if (!shmem->server.enabled) {
-        virBufferAsprintf(&buf, ",shm=%s,id=%s", shmem->name, shmem->info.alias);
+        virBufferAsprintf(&buf, ",shm=%s", shmem->name);
     } else {
-        virBufferAsprintf(&buf, ",chardev=char%s,id=%s", shmem->info.alias, shmem->info.alias);
+        virBufferAsprintf(&buf, ",chardev=char%s", shmem->info.alias);
         if (shmem->msi.enabled) {
             virBufferAddLit(&buf, ",msi=on");
             if (shmem->msi.vectors)
diff --git a/tests/qemuxml2argvdata/qemuxml2argv-shmem.args b/tests/qemuxml2argvdata/qemuxml2argv-shmem.args
index 89caf49..99fac11 100644
--- a/tests/qemuxml2argvdata/qemuxml2argv-shmem.args
+++ b/tests/qemuxml2argvdata/qemuxml2argv-shmem.args
@@ -17,19 +17,19 @@ QEMU_AUDIO_DRV=none \
 -no-acpi \
 -boot c \
 -usb \
--device ivshmem,shm=shmem0,id=shmem0,bus=pci.0,addr=0x3 \
--device ivshmem,size=128m,shm=shmem1,id=shmem1,bus=pci.0,addr=0x5 \
--device ivshmem,size=256m,shm=shmem2,id=shmem2,bus=pci.0,addr=0x4 \
--device ivshmem,size=512m,chardev=charshmem3,id=shmem3,bus=pci.0,addr=0x6 \
+-device ivshmem,id=shmem0,shm=shmem0,bus=pci.0,addr=0x3 \
+-device ivshmem,id=shmem1,size=128m,shm=shmem1,bus=pci.0,addr=0x5 \
+-device ivshmem,id=shmem2,size=256m,shm=shmem2,bus=pci.0,addr=0x4 \
+-device ivshmem,id=shmem3,size=512m,chardev=charshmem3,bus=pci.0,addr=0x6 \
 -chardev socket,id=charshmem3,path=/var/lib/libvirt/shmem-shmem3-sock \
--device ivshmem,size=1024m,chardev=charshmem4,id=shmem4,bus=pci.0,addr=0x7 \
+-device ivshmem,id=shmem4,size=1024m,chardev=charshmem4,bus=pci.0,addr=0x7 \
 -chardev socket,id=charshmem4,path=/tmp/shmem4-sock \
--device ivshmem,size=2048m,chardev=charshmem5,id=shmem5,msi=on,ioeventfd=off,\
+-device ivshmem,id=shmem5,size=2048m,chardev=charshmem5,msi=on,ioeventfd=off,\
 bus=pci.0,addr=0x8 \
 -chardev socket,id=charshmem5,path=/tmp/shmem5-sock \
--device ivshmem,size=4096m,chardev=charshmem6,id=shmem6,msi=on,vectors=16,\
+-device ivshmem,id=shmem6,size=4096m,chardev=charshmem6,msi=on,vectors=16,\
 bus=pci.0,addr=0x9 \
 -chardev socket,id=charshmem6,path=/tmp/shmem6-sock \
--device ivshmem,size=8192m,chardev=charshmem7,id=shmem7,msi=on,vectors=32,\
+-device ivshmem,id=shmem7,size=8192m,chardev=charshmem7,msi=on,vectors=32,\
 ioeventfd=on,bus=pci.0,addr=0xa \
 -chardev socket,id=charshmem7,path=/tmp/shmem7-sock
-- 
2.10.2

